### PR TITLE
fix: onerror

### LINKED ISSUE #4: (v2) Koa hangs on errors
Using this example from the README (modified with the `new` keyword for Koa 2), my server logs an error when I hit http://localhost:3000, but it does not respond and the request times out.

```javascript
'use strict'

var Koa = require('koa')
var logger = require('koa-pino-logger')

var app = new Koa()
app.silent = true //disable console.errors
app.use(logger())

app.use(function * () {
  this.body = 'hello world'
  throw Error('bang!')
})

app.listen(3000)
```

### PR DESCRIPTION
**Pull request by laggingreflex**
probably fixes #4 too

### CODE REVIEW

File: logger.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ function logger (opts, stream) {
   function pino (ctx, next) {
     wrap(ctx.req, ctx.res)
     ctx.log = ctx.request.log = ctx.response.log = ctx.req.log
-    ctx.onerror = catchErr(ctx, ctx.onerror)
+    ctx.onerror = catchErr(ctx, ctx.onerror.bind(ctx))
```
Comment (davidmarkclements) [inline logger.js]: I don't think the v8 which has `bind` optimized has landed in v8 yet, we should instead do `catchErr(ctx, function () { return ctx.onerror.apply(ctx, arguments) }))`

`bind` (before the it was optimized in later versions of v8) has notoriously bad performance

**Comment by ivan-kleshnin** [comment]
I confirm that this fixes #4 hang.

**Comment by davidmarkclements** [comment]
thank you! 

note: should benchmark normative try/catch solution now that it's been optimized in v8

**Comment by davidmarkclements** [comment]
note: broken because 4 support has dropped due to syntax

**Comment by davidmarkclements** [comment]
released 2.1.1

**Pull request merged**